Objects and object constructors lesson: Add article on this keyword (#24179)
TheOdinProject/curriculum: javascript/organizing_your_javascript_code/objects_and_object_constructors.md

Before we go much further, there's something important you need to understand about JavaScript objects. All objects in JavaScript have a `prototype`. Stated simply, the prototype is another object that the original object _inherits_ from, which is to say, the original object has access to all of its prototype's methods and properties.

> The concept of the prototype is an important one, so you’ve got some reading to do, which you'll find in the Assignment section below. Make sure you really get it before moving on!

Reviewer: I feel like the extra work is a non-issue. We are moving towards having our lessons have all the required readings in the assignment section, rather than sprinkled throughout the lesson.

If it helps, we also have a lesson navigation tool on every lesson page:
![image](https://user-images.githubusercontent.com/85733202/171996399-3bd8bd3c-bf49-4b39-84c1-62e03541cb7d.png)

### Assignment

<div class="lesson-content__panel" markdown="1">

1. Read up on the concept of the prototype from the articles below. 
   1. [This article](https://web.archive.org/web/20200513181548/https://javascriptissexy.com/javascript-prototype-in-plain-detailed-language/) is a straightforward introduction and demonstration of the concept. It also covers constructors again.. good time for a review! The important bits here, once you've covered the basics, are 'Prototype-based inheritance' and the 'Prototype chain'.
   2. To go a bit deeper into both the chain and inheritance, spend some time with [this great article](http://javascript.info/prototype-inheritance). As usual, doing the exercises at the end will help cement this knowledge in your mind. Don't skip them! Important note: This article makes heavy use of `__proto__` which is not generally recommended. The concepts here are what we're looking for at the moment. We will soon learn another method or two for setting the prototype.
2. You might have noticed us using the `this` keyword in object constructors and prototype methods in the examples above.

> Dmitri Pavlutin's excellent [article on the `this` keyword](https://dmitripavlutin.com/gentle-explanation-of-this-in-javascript/) is very comprehensive and covers how it changes in various situations. You should have a solid understanding of `this` after reading it. Pay special attention to the pitfalls mentioned in each section.

Reviewer: "This article", "this great article", and "article on the `this` keyword", are not ideal link texts for a11y reasons.

Less than ideal link text is present throughout our curriculum, but should be addressed soon. Although, let's improve the link texts here while we're updating this lesson.

For example, the second link text could be improved to:
```
... spend some time with [JavaScript.Info's article on Prototypal Inheritance](link_here). ...
```

Reviewer: > I understand that "This article" and "this great article" aren't ideal link texts and will improve them but I'd like to know what's wrong with "article on the this keyword" and how it could be improved in your opinion.

Yeah, "article on the this keyword" isn't bad for a link text. Although, it doesn't make a lot of difference but we could do something like "article on `this` keyword by Dimitri Pavlutin". 

If you're curious, you can read more about link text on [WebAIM's article on links and hypertext](https://webaim.org/techniques/hypertext/).

> Also, should I improve the link texts for the articles/videos in the "Additional Resources" section as well?

If you want to, but it's not necessary for this pull request.

- <a class="knowledge-check-link" href='https://javascript.info/prototype-inheritance'>Explain prototypal inheritance.</a>
- <a class="knowledge-check-link" href='#recommended-method-for-prototypal-inheritance'>Understand the basic do's and don't's of prototypical inheritance.</a>
- <a class="knowledge-check-link" href='https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create'>Explain what `Object.create` does</a>

> - <a class="knowledge-check-link" href='https://dmitripavlutin.com/gentle-explanation-of-this-in-javascript/'>Explain what the `this` keyword is and how it changes in various situations.</a>

Reviewer: Updating the knowledge check based on our wiki:
```suggestion
- <a class="knowledge-check-link" href='https://dmitripavlutin.com/gentle-explanation-of-this-in-javascript/'>How does `this` behave in different situations?</a>
```
